core/partitioning/shape_analysis.cpp

```diff
@@ -208,6 +208,8 @@ void getSegmentsOutputByRunning(
       jit_inputs_ivalues.push_back(ivalues_maps[input].toGenericDict());
     } else if (input->type()->kind() == torch::jit::TypeKind::DeviceObjType) {
       jit_inputs_ivalues.push_back(ivalues_maps[input].toDevice());
+    } else if (input->type() == (torch::jit::StringType::get())) {
```

Comment:
Consider switching to `input->type()->isSubtypeOf(torch::jit::StringType::get())`, for consistency with the above operators (lines 188 - 197)